crates/mun_compiler/src/diagnostics_snippets.rs

```diff
@@ -0,0 +1,372 @@
+use mun_hir::diagnostics::Diagnostic as HirDiagnostic;
+use mun_hir::{HirDatabase, HirDisplay};
+use mun_syntax::{
+    ast, AstNode, Parse, SourceFile, SyntaxError, SyntaxKind, SyntaxNodePtr, TextRange,
+};
+
+use std::sync::Arc;
+
+use mun_hir::line_index::LineIndex;
+
+use crate::annotate_snippets_builders::{AnnotationBuilder, SliceBuilder, SnippetBuilder};
```

Comment:
Maybe rename this to `crate::annotate` and include the `annotate_snippets::snippet::{AnnotationType, Snippet};` as `pub use`.